### ISSUE TITLE
editor not destroyed before page is unloaded

### ISSUE THREAD

**Issue by Tularis** [HstarComponents/ngx-ckeditor#11]
After leaving a page it will be unloaded by angular. If that page contained a ckeditor, it will automatically unload the DOM. This triggers an editor-destroy-iframe error from CKEditor (https://docs.ckeditor.com/ckeditor4/latest/guide/dev_errors.html#editor-destroy-iframe). Perhaps it should be destroyed explicitly within an ngOnDestroy method.

**Comment by Braste** [comment]
It already gets destroyed within an ngOnDestroy method, the problem is, that this doesn't happen fast enough and the dom gets changed by angular before ckeditor is able to destroy the iframe. If you want to prvent this error, you have to destroy it yourself in the ngOnDestroy method in your component.

**Comment by Tularis** [comment]
Actually I found that ngOnDestroy is always called *after* the DOM has been destroyed. See Issue [#14252](https://github.com/angular/angular/issues/14252) in angular. So we can't really fix this until angular is fixed.

I did notice however that if I comment the lines in OnDestroy of CKEditorComponent it will silence the exception, and thus make my application continuing in a functional state. We do leave a leak though... which is bad...
